src/pages/settings/users/edit/components/Permissions.tsx

```diff
@@ -119,6 +132,28 @@ export function Permissions(props: Props) {
       currentPermissions.push(permission);
     }
 
+    if (!currentPermissions.includes(`${permissionType}_all`)) {
+      const permissionsFromTypeLength = currentPermissions.filter(
+        (currentPermission) =>
+          currentPermission.startsWith(permissionType) &&
+          currentPermission !== 'view_reports' &&
+          currentPermission !== 'view_dashboard' &&
+          currentPermission !== 'disable_emails'
+      ).length;
+
+      if (permissionsFromTypeLength === permissions.length) {
+        currentPermissions = currentPermissions.filter(
+          (currentPermission) =>
+            !currentPermission.startsWith(permissionType) ||
+            currentPermission === 'view_reports' ||
+            currentPermission === 'view_dashboard' ||
+            currentPermission === 'disable_emails'
```

Comment:
These seems to repeat? Do you think it's worth extracting into separate array so we don't have to update both places?